### ISSUE TITLE
railway=station on area=yes behaves like a building, rather than landuse.

### ISSUE THREAD

**Issue by javbw** [gravitystorm/openstreetmap-carto#327]
When making many things in OSM, simply marking the point is okay, naming a building is good, and marking it's land use, then the objects inside (roads / buildings / walkways, etc) is best (as I understand it)

But there is no land use for station (as opposed to school, industrial, residential, etc) 

so the rendering of railway=station (and public_transport=station?) with area=yes should behave similarly to a land use tag - the area should accept things on above layers (pedestrian areas, platforms, etc) but it currently is does not. 

Please make the the railway=station tag behave in this manner.

**Comment by javbw** [comment]
comment continued from #1457 

I'm bringing this up because tagging it like other objects results in an unexpected action (there's no way to tag a station as a single landuse correctly). And the reasoning for unexpected action seems to not be in line with a lot of other tagging schemes for mapping areas of buildings and other places throughout town - so it causes confusion with new mappers (and me), and is counterintuitive (IMO) - espcially as landuse=* gets wider and wider (the last real holdout is somekind of landuse=civic, whcih Im working on in the tagging mailing list.) There are very few man-made/used areas that can't bedefined by a landuse. Finding stations like this is a real surprise. 

> The area in front of the station (bus stop, taxi booth, car parking, green space) does not fit this definition.

I understand about the landuse=railway not fitting the definition well (because, yea, a bus circle isn't railway), then why isn't there a landuse=transport_station or something similar? Should I suggest it in the tagging list?

Why are railway stations treated completely differently than any other place? It makes no sense - the tracks and the buildings are just as much of the station as the bus circle and other related amenities.   The *only* reason it is a station is because that is where the two come together. having a landuse for the railway side and a landuse for the non-railway side is a failure of the tagging system, IMO - the whole thing is one single place.

Having to tie pieces of things together to make a single whole thing feels very weird. No such weirdness is needed for an airport, though it is where two transportation methods collide (the parking lots, bus stations, and everything sit comfortably in "the airport's" landuse. The whole thing is a transport station. It also feels like this is against the "one thing, one object" that works so well for every other single building and landuse in OSM. With all the other landuses, I'm surprised that there isn't a good one for public transport stations.

I don't need a relation to add a driveway and parking to a mall. A station should be the same - especially with the "stop position" in existence to differentiate the different stop positions on the different lines (as I understand how relations could work).

**Comment by mboeringa** [comment]
> **Why are railway stations treated completely differently than any other place?** It makes no sense - the tracks and the buildings are just as much part of the station as the bus circle and other related amenities. **The platforms render in a completely unexpected manner (they render through buildings and disobey layer)**.

You probably don't want to hear this, but I think this is actually a ***conscious*** and ***deliberate*** choice by the developers of the styles (which I have implemented in my personal ArcGIS renderer too). By rendering tracks and platforms **on top** of all buildings, the layout of the railway station is visible, even in a 2D rendered map, without the need of fancy transparency, or 3D functionality.

This tells you way more about a station. For example, if the tracks were not drawn on top, and no transparency was applied, you would never be able to tell if the tracks were actually continuous through a station building, or a station possibly was a two sided "head" station.

Also, no information about track layout with switches and all, would be visible, as covered by the building.

Since there are many active "railway" mappers on OSM, I think they generally, or in the majority, prefer this type of rendering (but maybe not all like you...), versus one where nothing of the tracks is visible.

Transparency can only do so much... Personally, I find transparency quite useless with more than one transparent layer on top of each other, as it just becomes a mess of indefinable layers of which it unclear at what relative position they are, e.g. like here (Rotterdam CS, can you tell what is on top of the other?): 

![rotterdam_cs_mapquest](https://cloud.githubusercontent.com/assets/7635726/6821573/4ebacbb2-d2d8-11e4-9076-c4bce64dbaac.jpg)

In my renderer, to make the "tracks and platforms on top of buildings", more acceptable, I have decided to render the "building=roof" tag (for open sided roofs, often applied on railway stations) as an open cross hatch, which kind of gets you the best of both worlds: a roof over the tracks and platforms, but still visible tracks and platforms.

Also, the "building=roof" tagging, is often a more accurate measure of where the tracks are actually covered, as the building itself may be much bigger than the roof covering the tracks.

See this result:

![rotterdam_railway_station](https://cloud.githubusercontent.com/assets/7635726/5050143/0991c4ea-6c2c-11e4-9503-499104e354c4.jpg)

I also think you are seriously underestimating the difficulty of implementing, or doing something sensible, with **layer=x** or **level=x** in relation to buidlings, and even more so in combination with highways and railways. This is (hideously?...) complex to implement properly in combination with thematic rendering...

**Comment by math1985** [comment]
> The platforms render in a completely unexpected manner (they render through buildings and disobey layer). 

What is the rendering order you would expect?

**Comment by javbw** [comment]
mboeringa had a good comment - thanks for the replies. I think that whatever makes it better for parsing rail by itself is great for rail maps, but makes it difficult in default -carto to see where stations meet other transport - and giving you some idea of where to go if the station is complex. 

I am also out of my element completely, as I don't code, so what -carto creates is inside a back box to m, i can only comment on the output. 

i threw together a quick example of what I would expect the output to look like with railway=station on the area or a similar landuse=transport_station encompassing the station.  I did it from memory of shimmaebashi station  (the overtrack station), but it might be a little off. 
![trasparency](https://cloud.githubusercontent.com/assets/6212656/6822441/376c030e-d323-11e4-8bd3-580a9273ace7.png)

The station building is at ~80% transparancy.

**Comment by javbw** [comment]
Rendering the The roofs idea is a good one, but in this case the entire station is on posts over the tracks - convenience store, ramen shop, ticket office, bathrooms, everything is suspended over the tracks, so its more than a roof.

**Comment by javbw** [comment]
the station I'm talking about. https://goo.gl/maps/Kon3R  I want to be able to show that the station is over the tracks and platforms.  It is also hard to show that the overhead walkway directly accesses the station on the second floor (from way across the tracks from the north. 

Google maps has a user picture.  (no short link ><) https://www.google.com/maps/place/Gunma+Prefecture,+Japan/@36.378902,139.04711,3a,75y,90t/data=!3m5!1e2!3m3!1s-k_qFkFZ9NS8%2FUqz83yK_ZII%2FAAAAAAAAAng%2Fm3xxhgT8qFw!2e4!3e12!4m2!3m1!1s0x601e6085c836ae59:0xd799356889717292

**Comment by kocio-pl** [comment]
I may be wrong, but I guess this is similar problem I described about main stations in Berlin or Warsaw (the roof is the most important shape here). See https://github.com/gravitystorm/openstreetmap-carto/issues/688 (also https://github.com/gravitystorm/openstreetmap-carto/issues/552 to some extent): the more complex building with more levels, the more issues with current layering. My opinion is this belongs rather to the public transport map than to the general one, although it would be just a quick fix - the final form should be linking to indoor/3D models.

**Comment by javbw** [comment]
@mboeringa - I reread your comment again. The transparency issues are difficult in crowded environments. by looking at the tracks, I guess the smaller station is under the larger station in your example. a more detailed mapping, of the surrounding environment might provide for more visual clues.  This will probably have implications for the stations more than the platforms or the tracks - I still think the idea of having the platforms follow layer is a good one if roofs are dealt with is some translucent way - your crosshatch looks really good. There still needs to be some very soft transparency allowed for the platforms, as some stations are very large (and more than a roof, as they sit over the tracks), but t least having the platforms and to roofs play nicer with each other would be a start. 

~~

Access to the station itself should be clearer than than the platforms in default -carto, as how to actually get in the station and where the station is can be more easily inferred if the buildings are rendered with more priority than the platforms. A light transparency and good roof rendering will also allow for understanding the platforms on closer inspection.

I'm not sure of the exact solution, but treating stations as something more than a building and a place with a landuse is odd - especially since there are other pieces (stop position, building=station) than can take up the slack. The layer and transparency issues with the tracks and platforms exacerbate the issue.

**Comment by Andygol** [comment]
@matkoniecz This is not good solution https://github.com/gravitystorm/openstreetmap-carto/issues/327#issuecomment-58996478

I faced with this [weird subway platform rendering](http://www.openstreetmap.org/way/264918883/history#map=17/50.44552/30.51634) with **layer=-1** + **public_transport=platform** + **railway=platform**, so here should take into account a value of layer tag
![image](https://cloud.githubusercontent.com/assets/369696/10700293/64a73264-79c5-11e5-863f-9621325f6481.png)

**Comment by kocio-pl** [comment]
> building=train_station should only be tagged around railway buildings, if any, and rendered like an important building

I guess this is not enough for a station building to be rendered as important. The building can be currently used for some other things (like cafe near Warsaw), so we need to have it tagged also with railway=station.

**Comment by dieterdreist** [comment]
sent from a phone

> Il giorno 25 set 2016, alle ore 03:49, kocio-pl <notifications@github.com> ha scritto:
> 
> The building can be currently used for some other things (like cafe near Warsaw), so we need to have it tagged also with railway=station.


you could check for a building=trainstation inside a railway=station area. If you tag the building with the railway tag, how do you tag the whole station? Do you have a railway=station inside a railway=station area?

**Comment by kocio-pl** [comment]
It all depends on how do we understand "railway=station"?

If we just read it on wiki, there's only one way - there may be only one such tag, so this must be used for the widest scope only.

But wiki is not set in stone - maybe nobody thought about the case when the station building is not used like it was intended? Then it makes a lot of sense to me to tag **station area** as railway=station, and **station building** as building=* (whatever form it is, usually a train_station of course) and railway=station (because this is the current function of this building).

In the end I think it's safer to go with  rendering all the building=train_station as major, because "major" is just what we want it to be. See [my PR](https://github.com/gravitystorm/openstreetmap-carto/pull/2371#issuecomment-249412951).

**Comment by javbw** [comment]
I had commented on this because I want to define the station's area to encompass all the amenities. 

I may have a wrong idea about tagging, but in the same way we tag landuse=retail to define a shopping mall and it's amenities, and building=retail to define the structures, I want to define the area used by a station and it's amenities, especially when some amenities (entrances, walkways, driveways, parking) is adjacent to places where you cannot park or walk or access as a traveler. 

The example I wanted to map is to avoid this kind of problem that google maps has: 

https://goo.gl/maps/xgX2jHq1kk22
Google maps the platforms, but not the station (the building to the NW of the platforms). It is impossible to access the platforms from the south side or west side whatsoever, even along the tracks, it all has to be accessed through the north through the station. 

We have the building=train_station and railway=platform, and the promenade sublet for the walkways, but in this case, they are separate things. There is also diffferent entrances, parking and bus loops, all an amenity to the station (not the rail tracks, sidings, or other landuse=railway, but the amenities of the station itself)

On OSM 
https://www.openstreetmap.org/#map=18/36.42592/139.27667



Just as we can map a store as a point, a building, or a collection of amenities and buildings on a landuse, mapping should be consistent above all else, and the historical or architectural significance of a train station not used as a train station should be tagged and rendered differently (I don't want a train museum in an old station building that is no longer a transportation stop to ever be rendered as an actual working station), especially at the expense of consistent mapping. 

Javbw

**Comment by kocio-pl** [comment]
> I had commented on this because I want to define the station's area to encompass all the amenities.

Both ways respect this rule.

> the historical or architectural significance of a train station not used as a train station should be tagged and rendered differently

But how? While I like your rants, because they are always insightful, IMO It's time to talk about details rather than just general ideas, at least here. I understand that your issue is not about station buildings, but currently it doesn't look to me like it's going to be solved easily and many topics discussed here should be on Tagging list really before we get back here.

**Comment by javbw** [comment]
> On 26 Sep 2016, at 10:06 AM, kocio-pl <notifications@github.com> wrote:
> But how?


We use theme=* for playgrounds, and we use foobar:disused=* & foobar:abandoned=* to denote a previous existing business or function of a building. 
Perhaps
 foobar:previous=* 
 foobar:architecture=* 
foobar:built_as=*

People have cafes in old train cars, planes, and nearby my house here in Japan is an old double-decker London bus used as a cafe. It is part of the ground now. 

Building=retail 
Shop=cafe
Building:built_as=bus 

I don't think we would pull public_transport into it because it is a bus, nor should building=train_station (if it denotes a building *currently used* as a station) should be used as an old station. 

This would let Building=train_station work for the building, and railway=station be used for the area of its amenities- the area where the railway interacts with other transportation methods, separating it from land use=railway. 

There are hundreds of repurposed 7-11s in Japan. You know instantly it was a 7-11 when seeing it as you drive by, now used as a laundromat or a ramen shop. 

Some have been turned into offices. 
Would I still use building=retail when it clearly is a building=commercial or =office now?  Or does the fact that it once was a 7-11 change its building classification? I think not. Same is true with train station. Moving the architecture away from usage, especially useful "major" building categories would allow for proper and consistent tagging, instead of train stations being weird outliers compared to other buildings and complexes that are easier to map and logically understand the nesting relationship between building, amenity, and landuse keys. 

Javbw.

**Comment by dieterdreist** [comment]
sent from a phone

> Il giorno 26 set 2016, alle ore 04:41, javbw <notifications@github.com> ha scritto:
> 
> Building=retail 
> Shop=cafe
> Building:built_as=bus


discuss tags on tagging, this is an issue tracker for the rendering style and an issue about train stations.

a bus used as cafe is far from a building=retail. 
building=bus + amenity=cafe
building is about "built as", no need for new tags

**Comment by javbw** [comment]
> On 26 Sep 2016, at 6:13 PM, dieterdreist <notifications@github.com> wrote:
> 
> discuss tags on tagging
This seems to be something that will or will not be implemented in the renderer because of tagging philosophy and practices. Is there a better place where both can mingle? 

Im sorry my bus example is incorrect. I will follow any guidelines that are suggested in the wiki. 

however the 7-11 example is a perfect example of how building=* defining architecture can be problematic, rather than purpose. 

~~~~~

I want the ability to define a station as an area - just as many other building / manmade tags have an associated area to define large and complex places. I thought railway=station would work, as building=railway_station exists, allowing for nested use, yet separates it from landuse=railway, which is for any land used for rail alignments. 

I will refrain from commenting further, unless directly addressed, as I feel my points are understood. I hope a solution can be found. 

Javbw.

**Comment by kocio-pl** [comment]
@javbw I believe you should change the name of this issue or close it and start a new one (whatever you feel is right), because now it's hard for me to understand what is the scope in the end.

**Comment by javbw** [comment]
I think the title explains it the best. 

If I put railway=station on a polygon, the renderer assumes it is a building, even though we have building=train_station for further differentiation. So if I know the boundary of a train station area, and draw a polygon to show the land used by the station (its bus circle, its bike parking, its multiple related buildings and amenities that all belong to the station, as I would a landuse) the polygon is rendered as if it is a building, even with an area=yes tag.

I want railway=station+area=yes to be rendered like other *areas*, with ways and building polygons able to be rendered over it (respecting the layer=tag), rather than as a building that always renders over everything.

If this means it needs to have some other tag pairing(s), such as building=no / layer=0 / landuse=railway for the renderer to correctly understand that it shouldn't be rendered as a building - great, I will add those tags if deemed necessary by you guys. I just want a rendered "area" that is not the same shade and strength as a station building, so the station building on this new area is still identifiable as a building, yet no ambiguity that the area around the building belongs to the station too (like a school campus or airport grounds). 

I know my explanations are a bit wordy, but I am trying to be succinct. 

Javbw

**Comment by kocio-pl** [comment]
It's still to general for me. =}

> renderer assumes it is a building,

For me it looks like the area, not like building:
http://www.openstreetmap.org/way/114507529

If it's only about layers, its a more general issue (like "make layers work global for all type of objects").

Did I understand you correctly?

**Comment by javbw** [comment]
> On Sep 29, 2016, at 11:51 PM, kocio-pl <notifications@github.com> wrote:
> 
> It's still to general for me. =}
> 
> renderer assumes it is a building,
> 
> For me it looks like the area, not like building:
> http://www.openstreetmap.org/way/114507529 <http://www.openstreetmap.org/way/114507529>
> If it's only about layers, its a more general issue (like "make layers work global for all type of objects").
> 
> Did I understand you correctly?
> 
> 

Here is an area I just drew around Ota train station with railway=station + area=yes  This is the extent of the station - the area that where the road, bus, bicycle, and pedestrian system meets the train system (“the station”), *and* is unambiguously and easily mappable from imagery as “the station” *and* is the extent of the area that contains the amenities owned and operated by “the station”.  the area around the station is not fully nor correctly mapped currently (no pedestrian areas at all, no station amenities nor surrounding businesses), but this polygon is pretty accurate (to about the meter). this area rendered in the darker red like a building. I consider this unexpected behavior, as it is not a building. 

Contrast this to to convenience store near the south circle: the retail landuse with a retail building and a pin. they render as expected. 

I have totally mapped Maebashi station, https://www.openstreetmap.org/#map=19/36.38331/139.07330 <https://www.openstreetmap.org/#map=19/36.38331/139.07330> and this is when this rendering issue really bothered me. 

Between railway=station rendering like a red building (and building=train_station not rendered in red for some reason) and highway=pedestrian ignoring or over-rendering roof polygons or not rendering covered=yes, it was not rendered in a way I expected. The giant dark red ground flooding the bus circles looked very bad in particular. I removed the railway=station tag from the area and made this ticket because it rendered in an unexpected way that would be a misrepresentation of the station.  Not having a proper accompanying light red to go with the “station building red”  does not accurately depict the station, and not mapping the extent at all is an oversight. So I expect, like many other tags, adding area=yes to the polygon will tell the renderer that this is an area, and treat it as an area, not a way nor a building. 

railway=station+area=yes should be rendered in a shade that is useful as a landuse, rather than “strong building red”.

for stations where the platforms and the station buildings are are not on top of each other, and the area covered by the station’s other amenities greatly exceeds the area covered by the just the platforms and station building, then the current rendering is not useful and doesn’t render in an expected manner (to me). 

Does that make it clear?

Javbw

**Comment by kocio-pl** [comment]
It was still not succinct for me, but I see several problems here:

1. Tagging station is not clear enough currently - thanks for [starting discussion on Tagging list](https://lists.openstreetmap.org/pipermail/tagging/2016-October/030301.html)!

2. Station color is too dark for you. I try to use it for [rendering ferry terminal](https://github.com/gravitystorm/openstreetmap-carto/pull/2385), because it's darker than airport area, but we could make it lighter, as we did lately with few others. Any color you'd like to have here?

3. Station building could be rendered [like major building](https://github.com/gravitystorm/openstreetmap-carto/pull/2371) - and I just have technical difficulty to do it, any help would be welcome.

4. Layers belong to other issues, like https://github.com/gravitystorm/openstreetmap-carto/issues/688.

What do you think about these descriptions and maybe is there something still missing? It's better to stick with small, well-defined issues, because they have much bigger chances to be solved.

**Comment by javbw** [comment]
Javbw

> On 3 Oct 2016, at 4:53 AM, kocio-pl <notifications@github.com> wrote:
> 
> It was still not succinct for me, but I see several problems here:
> 
> Tagging station is not clear enough currently - thanks for starting discussion on Tagging list!
> 
No problem, I'm trying to understand if my assumptions on tagging and how it affects rendering decisions. Seems that I am making bad assumptions because I am confusing wildly inconsistent tagging philosophies between different tags, leading to this thread. 


> Station color is too dark for you. I try to use it for rendering ferry terminal, because it's darker than airport area, but we could make it lighter, as we did lately with few others. Any color you'd like to have here?
> 
I was under the assumption that the "brick" red color was for station *buildings* to make them extremely visible. Is that dark red color meant for station area? This might be the source of my confusion and this misunderstanding.  I assumed people were tagging this on Buildings, making the buildings dark red (similar to the Google maps rendering). Is this dark red color the proper color for non-building areas(like a landuse)?? 

Airports are a perfect example:

terminal buildings: dark purple.
Runways/taxiways/apron: grey
Overall airport area: light purple 

So I want train stations to be rendered this consistently as well: 

Train Station buildings: dark red
Platforms: dark grey 
Overall station area: ?????? [this thread] 


> Station building could be rendered like major building - and I just have technical difficulty to do it, any help would be welcome.
> 
Because I don't use OSM in a place where it is well mapped, both in density of ways & areas as well as tagged information (Japan is very sparsely mapped and tagged correctly, much is old or outdated tagging outside of Tokyo or a scrambled mess from a GIS import 6 years ago) maybe my idea of what a "station" is - the land, the ped access, and the building renders might be outdated. Aren't stations rendered as the most visible and outstanding buildings  (the red color) already?  
> Layers belong to other issues, like #688.
> 
Yea, sorry, shouldn't have brought it up. It's part of my frustration of mapping large complexes, but it is not part of this ticket. 
> What do you think about these descriptions and maybe is there something still missing? It's better to stick with small, well-defined issues, because they have much bigger chances to be solved.
> 
It seems form a few comments on the mailing list that my assumption that the area=yes tag on the area would somehow signal that it is not a building is not right. Perhaps I should be looking to add building=no and have that tell the renderer that is an area and not a building. But I am looking for that light area, like the light purple airport grounds. 

This might close this whole thread, I'm sorry if my incorrect assumptions have wasted your time. 

Javbw

**Comment by kocio-pl** [comment]
> I was under the assumption that the "brick" red color was for station *buildings* to make them extremely visible. Is that dark red color meant for station area? This might be the source of my confusion and this misunderstanding. 

I don't know, maybe it was back then. In https://github.com/gravitystorm/openstreetmap-carto/issues/389 it was not clear. 

But since I prefer to make station building to be rendered as a major building (dark brown, like in building+place of worship) it doesn't matter now and we can focus on area color.

**Comment by kocio-pl** [comment]
Area option for railway=station [has been restored](https://wiki.openstreetmap.org/w/index.php?title=Tag%3Arailway%3Dstation&type=revision&diff=1353666&oldid=1340922) on the wiki (it was like this until last year, as someone wrote in the Tagging thread).

**Comment by kocio-pl** [comment]
The discussion on Tagging keeps going, so I hope we will have consistent proposition in the near future, which will resolve all our problems with railway stations.

**Comment by jojo4u** [comment]
> which will resolve all our problems with railway stations.

Giving the nature of OSM this is meant as a joke? :)

**Comment by kocio-pl** [comment]
Well, I said _our_ problems (related to rendering), not _all_ of the OSM problems. :smile:

**Comment by dieterdreist** [comment]
2016-10-05 15:58 GMT+02:00 kocio-pl <notifications@github.com>:

> Well, I said *our* problems (related to rendering), not *all* of the OSM
> problems.
>

there's a tiny detail apparently getting in the way (until hstore gets
activated or missing keys added), that likely it would be desirable to
place the icon according to the public_transport=station tag, which is
focussing on passengers, and which will probably be covering only the parts
of a station that are interesting to the passenger (buildings, accessible
ways, platforms)

**Comment by kocio-pl** [comment]
Now that railway station buildings render as major buildings, I want to change railway station area color to be the same as aeroway=aerodrome. It's part of a bigger transport areas unification plan, see https://github.com/gravitystorm/openstreetmap-carto/issues/2094.

See for example this area with railway station building inside:
http://www.openstreetmap.org/way/114507529

**Comment by Tomasz-W** [comment]
@kocio-pl Rendering of this feature seems to be ok (there is a label and a fill also). 
See:
https://www.openstreetmap.org/way/114507529
https://www.openstreetmap.org/way/479178948

Shouldn't it be closed?

**Comment by kocio-pl** [comment]
I think you're right - thanks for checking!
